tests/test_datasets.py

```diff
@@ -14,7 +14,7 @@
 class TestDataset(object):
     def setUp(self):
         self.data = [1, 2, 3]
-        self.stream = DataStream(IterableDataset(self.data))
+        self.stream = IterableDataset(self.data).get_example_stream()
```

Comment:
I'd be inclined to have a convention along the lines of "if you instantiate `DataStream` without giving it an iteration scheme, it'll take the dataset's `example_iteration_scheme` by default"